tripal/tripal
# Fixes to load EDAM ontology.

# Bug Fix
Issue #1116

## Description


It is not possible to load the EDAM vocabulary into Chado using the Tripal OBO loader for the reasons mentioned in the issue linked above.   Most of them are issues with the formatting of the EDAM OBO and the organization of the EDAM vocabulary. One is a limitation in the OBO loader.

Unfortunately, because the EDAM OBO does not follow the standard properly some of the fix involved hardcoding some cleaning steps that are unique to EDAM.  I didn't like doing this, but I think it's the easiest way to deal with it.

## Testing?





1. Before pulling the code for this PR, add the EDAM vocabulary at Admin >> Tripal >> Data Loaders >> Chado Vocabularies >> OBO Vocabulary Loader.  Use the following 
   - Vocabulary Name:  EDAM
   - Remote URL:  http://edamontology.org/EDAM.obo
2.  Submit a job to import the EDAM vocabulary. You should get an error.
3. Pull this code.
4.  Re-submit a job to import the EDAM vocabulary. It should go in.
5.  Inspect your database tables.   
     - The EDAM record in the `cv` table should already exist because it was loaded by the Chado Prepare step.  the EDAM cv record will not the the highest cv_id value.
     - The `format`, `data`, `operation` and `topic` db records  should already exist because they are loaded with a Chado prepare.
     - Browse your Tripal site using these relative paths and check that the terms were uploaded for the vocabulary:  
        - `/cv/lookup/data`
        - `/cv/lookup/format`
        - `/cv/lookup/operation`
        - `/cv/lookup/topic`

## Screenshots (if appropriate):

## Additional Notes (if any):
